mmv1/third_party/terraform/services/dataflow/resource_dataflow_flex_template_job.go.erb

```diff
@@ -841,4 +814,64 @@ func dataflowFlexJobTypeParameterOverride(ename, pname string, d *schema.Resourc
 	return nil
 }
 
+func hasIllegalParametersErr(d *schema.ResourceData) error {
+	pKey := "parameters"
+	errFmt := "%s must not include Dataflow options, found: %s"
+	for k := range ResourceDataflowFlexTemplateJob().Schema {
+		if _, notOk := d.GetOk(fmt.Sprintf("%s.%s", pKey, k)); notOk {
+			return fmt.Errorf(errFmt, pKey, k)
+		}
+		kk := toCamelCase(k)
+		if _, notOk := d.GetOk(fmt.Sprintf("%s.%s", pKey, kk)); notOk {
+			return fmt.Errorf(errFmt, pKey, kk)
+		}
+	}
+	return nil
+}
+
+func toCamelCase(s string) string {
+	var result []rune
+	runes := []rune(s)
+	for i := 0; i < len(runes); i++ {
+		r := runes[i]
+		if i == 0 {
+			result = append(result, r)
+			continue
+		}
+		if r != '_' {
+			result = append(result, r)
+			continue
+		}
+		i++
+		r = runes[i]
+		result = append(result, unicode.ToUpper(r))
+	}
+
+	return string(result)
+}
```

Comment:
I'm finding this function a bit hard to read. Could we use [SnakeToPascalCase](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tpgresource/utils.go#L593-L599) from the `tpgresource` package instead?

Reply:
@trodge Thank you for the tip. Done :-)